File: BarberBooking.API/Filters/Salon/SalonFilter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using BarberBooking.API.Domain.ValueObjects;
using Microsoft.EntityFrameworkCore.Storage.ValueConversion;

namespace BarberBooking.API.Filters
{
    public class SalonFilter
    {
        public bool? IsActive {get; set;}
        public bool? MaxRating {get; set;}
        public bool? Popular {get; set;}
        public bool? MinPrice {get; set;}
    }
}

File: barber_booking_app/lib/models/params/salon_params/salon_filter.dart
import 'package:barber_booking_app/models/base/json_serializable.dart';
class SalonFilter extends JsonSerializable {
  bool? IsActive;
  bool? MaxRating;
  bool? Popular;
  bool? MinPrice; 
  SalonFilter({this.IsActive , this.MaxRating, this.Popular, this.MinPrice});
  
  @override
  Map<String, dynamic> toJson() {
    return{
      'isActive':IsActive,
      'maxRating':MaxRating,
      'popular':Popular,
      'minPrice':MinPrice
    };
  }
}